routers/web/user/package.go

```diff
@@ -449,37 +454,42 @@ func PackageSettingsPost(ctx *context.Context) {
 	form := web.GetForm(ctx).(*forms.PackageSettingForm)
 	switch form.Action {
 	case "link":
-		success := func() bool {
-			repoID := int64(0)
-			if form.RepoID != 0 {
-				repo, err := repo_model.GetRepositoryByID(ctx, form.RepoID)
-				if err != nil {
-					log.Error("Error getting repository: %v", err)
-					return false
-				}
-
-				if repo.OwnerID != pd.Owner.ID {
-					return false
-				}
-
-				repoID = repo.ID
+		if form.RepoFullName == "" { // remove the link
+			if err := packages_model.SetRepositoryLink(ctx, pd.Package.ID, 0); err != nil {
+				ctx.Flash.Error(ctx.Tr("packages.settings.unlink.error"))
+			} else {
+				ctx.Flash.Success(ctx.Tr("packages.settings.unlink.success"))
 			}
+			ctx.JSONRedirect("")
+			return
+		}
 
-			if err := packages_model.SetRepositoryLink(ctx, pd.Package.ID, repoID); err != nil {
-				log.Error("Error updating package: %v", err)
-				return false
+		owner, name, _ := strings.Cut(form.RepoFullName, "/")
+		repo, err := repo_model.GetRepositoryByOwnerAndName(ctx, owner, name)
```

Comment:
Where is the permission check? I can link a package to any "owner/repo"?

Reply:
And if the owner can't be changed, why you added the "full name" to the repo list?